geopandas/geodataframe.py

```diff
@@ -810,6 +810,9 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         ids = np.array(self.index, copy=False)
         geometries = np.array(self[self._geometry_column_name], copy=False)
 
+        if not self.columns.is_unique:
+            raise ValueError("DataFrame columns must be unique")
```

Comment:
```suggestion
            raise ValueError("GeoDataFrame cannot contain duplicated column names.")
```

This feels a bit more intuitive to me.